src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -450,6 +448,69 @@ void OTARequestor::NotifyUpdateApplied(uint32_t version)
     ConnectToProvider(kNotifyUpdateApplied);
 }
 
+CHIP_ERROR OTARequestor::GetDefaultOtaProviderList(AttributeValueEncoder & encoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    return encoder.EncodeList([&](const auto & enc) -> CHIP_ERROR {
+        mDefaultOtaProviderList.ForEachActiveObject([&](ProviderLocation::Type * pl) {
+            err = enc.Encode(*pl);
+            VerifyOrReturnError(err == CHIP_NO_ERROR, Loop::Break);
+            return Loop::Continue;
+        });
+        return err;
+    });
+}
+
+CHIP_ERROR OTARequestor::ClearDefaultOtaProviderList(FabricIndex fabricIndex)
+{
+    // Remove all entries for the fabric index indicated
+    mDefaultOtaProviderList.ForEachActiveObject([&](ProviderLocation::Type * pl) {
+        if (pl->GetFabricIndex() == fabricIndex)
+        {
+            mDefaultOtaProviderList.ReleaseObject(pl);
+        }
+        return Loop::Continue;
+    });
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OTARequestor::AddDefaultOtaProvider(FabricIndex fabricIndex, ProviderLocation::Type const & providerLocation)
+{
+    if (providerLocation.GetFabricIndex() == kUndefinedFabricIndex)
```

Comment:
We should be enforcing this in the IM code.  If we're not, please file an issue on that?

Reply:
I see that even if I specify an invalid fabric index on the client side, it always gets converted to the accessing fabric index. I'll remove this check.